### PR TITLE
Teach DGModel about horizontal and vertical

### PR DESCRIPTION
**Pull request by lcw**
This teaches `DGModel` to apply its derivatives in the horizontal, the vertical, or every direction.

### CODE REVIEW

**Comment by lcw** [comment]
bors try

File: test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion.jl [hunk 1]
```diff
@@ -175,63 +179,109 @@ let
...
+          β = FT(1 // 100)
+          μ = FT(-1 // 2)
+          δ = FT(1 // 10)
+          for l = 1:numlevels
+            Ne = 2^(l-1) * base_num_elem
+            brickrange = ntuple(j->range(FT(-1); length=Ne+1, stop=1), dim)
+            periodicity = ntuple(j->false, dim)
+            topl = StackedBrickTopology(mpicomm, brickrange;
+                                        periodicity = periodicity)
+            @show dt = (α/4) / (Ne * polynomialorder^2)
```
Comment (mwarusz) [inline test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion.jl]: Unrelated to this PR but maybe use this occasion to change this `show` to `info`.

Comment (lcw) [reply test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion.jl]: Good idea.  I pushed out this change.

**Comment by lcw** [comment]
bors try

**Comment by lcw** [comment]
bors try

File: test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion.jl [hunk 2]
```diff
@@ -175,63 +179,109 @@ let
...
+            brickrange = ntuple(j->range(FT(-1); length=Ne+1, stop=1), dim)
+            periodicity = ntuple(j->false, dim)
+            topl = StackedBrickTopology(mpicomm, brickrange;
+                                        periodicity = periodicity)
+            @info dt = (α/4) / (Ne * polynomialorder^2)
+
+            timeend = 1
+            outputtime = 1
+
+            dt = outputtime / ceil(Int64, outputtime / dt)
```
Comment (simonbyrne) [inline test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion.jl]: `dt` is not defined on the LHS

File: test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion.jl [hunk 3]
```diff
@@ -175,63 +179,109 @@ let
...
+          β = FT(1 // 100)
+          μ = FT(-1 // 2)
+          δ = FT(1 // 10)
+          for l = 1:numlevels
+            Ne = 2^(l-1) * base_num_elem
+            brickrange = ntuple(j->range(FT(-1); length=Ne+1, stop=1), dim)
+            periodicity = ntuple(j->false, dim)
+            topl = StackedBrickTopology(mpicomm, brickrange;
+                                        periodicity = periodicity)
+            @info dt = (α/4) / (Ne * polynomialorder^2)
```
Comment (simonbyrne) [inline test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion.jl]: @mwarusz pointed out that this is evaluated in its own local scope (unlike `@show`)
```suggestion
            dt = (α/4) / (Ne * polynomialorder^2)
            @info "time step" dt
```

**Pull request merged**